### PR TITLE
Use kinematics plugin instead of inverse Jacobian for servo IK

### PR DESCRIPTION
**Pull request by wyattrees**
### Description

This changes servo to utilize a kinematics plugin if one is defined. If one is not present or doesn't support the group, default to inverse Jacobian

Also updated servo_example launch file to use MoveItConfigsBuilder.

To test, the current moveit_servo tutorial can be used as is. The modification of the servo_example launch file in this repository also includes a reference to kinematics.yaml for the servo node.

The kinematics plugin used by the servo node is defined via ROS parameters, which may be loaded from a yaml file. If using `MoveItConfigsBuilder` in your servo launch file (as is done in `servo_example.launch.py` in this PR), you can pass the necessary parameters as such:

```python
moveit_config = (
        MoveItConfigsBuilder("moveit_resources_panda")
        .robot_description(file_path="config/panda.urdf.xacro")
        .to_moveit_configs()
    )

servo_node = Node(
        package="moveit_servo",
        executable="servo_node_main",
        parameters=[
            servo_params,
            moveit_config.robot_description,
            moveit_config.robot_description_semantic,
            moveit_config.robot_description_kinematics, # here is where kinematics plugin parameters are passed
        ],
```

In the above example, the `kinematics.yaml` file is taken from the `moveit_resources` repository in the workspace, specifically `moveit_resources/panda_moveit_config/config/kinematics.yaml`. By default, this yaml file defines the KDL plugin to be used. To use `bio_ik` as the solver, first clone the `ros2` branch of [the repository](https://github.com/PickNikRobotics/bio_ik) into your workspace. Then edit the above yaml file to contain the following:

```yaml
panda_arm:
  kinematics_solver: bio_ik/BioIKKinematicsPlugin
  kinematics_solver_search_resolution: 0.005
  kinematics_solver_timeout: 0.05 # this timeout parameter is actually ignored by servo
  kinematics_solver_attempts: 1.0
  mode: gd_c # see bio_ik for other modes, but gd_c (gradient descent) has the best performance
```
The timeout parameter is ignored for servo, as it uses a fraction of the publish period instead.

The actual ROS parameter names that get passed by loading the yaml file are of the form `robot_description_kinematics.<group_name>.<param_name>`, e.g. `robot_description_kinematics.panda_arm.kinematics_solver`

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by wyattrees** [comment]
@AndyZe I've added more to the description showing how to change the kinematics plugin.

File: moveit_ros/moveit_servo/launch/servo_example.launch.py
Comment (wyattrees) [reply moveit_ros/moveit_servo/launch/servo_example.launch.py:109]: Sure thing -- is it commented out because, as of now, it doesn't work as a `ComposeableNode`? Should it just be removed (in a separate PR)?

Comment (AndyZe) [reply moveit_ros/moveit_servo/launch/servo_example.launch.py:109]: I think somebody should test if it works as a `ComposableNode` yet, then delete it if not. Doesn't have to be you right now.

Comment (AndyZe) [reply moveit_ros/moveit_servo/launch/servo_example.launch.py:109]: issue #1441

File: moveit_ros/moveit_servo/src/servo_calcs.cpp
Comment (AndyZe) [reply moveit_ros/moveit_servo/src/servo_calcs.cpp:604]: We discussed this and `static` doesn't fit here because it needs to be initialized to the identity matrix with every iteration.

File: moveit_ros/moveit_servo/src/servo_calcs.cpp
Comment (AndyZe) [inline moveit_ros/moveit_servo/src/servo_calcs.cpp:622]: Sorry to backtrack. From reading here, `static` does not really help in terms of realtime performance. Please remove `static`.  :P

https://stackoverflow.com/a/3835902

File: moveit_ros/moveit_servo/src/servo_calcs.cpp
Comment (AndyZe) [inline moveit_ros/moveit_servo/src/servo_calcs.cpp:630]:
```suggestion
        delta_theta_.at(i) = solution.at(i) - internal_joint_state_.position.at(i);
```

File: moveit_ros/moveit_servo/src/servo_calcs.cpp
Comment (AndyZe) [inline moveit_ros/moveit_servo/src/servo_calcs.cpp:628]:
```suggestion
      for (size_t i = 0; i < num_joints_; ++i)
```

**Review by AndyZe — COMMENTED** [review]
Sorry for these very nitpicky changes. LGTM after this.

**Review by AndyZe — APPROVED** [review]

**Comment by wyattrees** [comment]
Found a bug in how rotations are applied while doing some singularity avoidance testing, we should hold off on merging for now.

**Comment by wyattrees** [comment]
@AndyZe I've fixed an issue where rotations were always being applied in the same frame--would you mind taking a second look at the recent changes (lines 613-633)?

File: moveit_ros/moveit_servo/src/servo_calcs.cpp
Comment (AndyZe) [inline moveit_ros/moveit_servo/src/servo_calcs.cpp:613]: I think these comments ought to be more generic. It's not necessarily the tip link reference frame, but rather whatever the IK reference frame is set to, right?

Otherwise the feature of switching from World to EE frame as the user's frame of reference (mentioned in [the tutorial](https://moveit.picknik.ai/humble/doc/examples/realtime_servo/realtime_servo_tutorial.html)) would not work.

Comment (wyattrees) [reply moveit_ros/moveit_servo/src/servo_calcs.cpp]: Yeah, it really should be more generic -- but it's actually more than just the comment. Really, we shouldn't just blindly use `tf_moveit_to_ee_frame_`, but rather a transform from what the IK solver knows as the base frame -> what the IK solver knows as the tip frame (which is likely the same as what servo knows as moveit -> ee frame, but not necessarily). 

Switching of the reference frame for the incoming twist commands is already handled above in the same function, so either way it should work for any incoming frame of reference.

See the latest commit for my solution to this

**Review by AndyZe — APPROVED** [review]

**Pull request merged**
